ui/page/odyseeMembership/view.jsx

```diff
@@ -0,0 +1,613 @@
+    const purchaseString = buildPurchaseString(price.unit_amount, price.recurring.interval, planName);
+
+    openModal(MODALS.CONFIRM_ODYSEE_MEMBERSHIP, {
+      membershipId,
+      userChannelClaimId: noChannelsOrIncognitoMode ? undefined : userChannelClaimId,
+      userChannelName: noChannelsOrIncognitoMode ? undefined : userChannelName,
+      priceId,
+      purchaseString,
+      plan: planName,
+      populateMembershipData,
+      setMembershipOptions,
+      updateUserOdyseeMembershipStatus,
+      user,
+    });
+  };
+
+  const cancelMembership = async function (e, membership) {
+    const membershipId = e.currentTarget.getAttribute('membership-id');
+
+    const cancellationString =
+      'You are cancelling your Odysee Premium Membership. You will still have access to all the paid ' +
+      'features until the point of the expiration of your current membership, at which point you will not be charged ' +
+      'again and your membership will no longer be active.';
+
+    openModal(MODALS.CONFIRM_ODYSEE_MEMBERSHIP, {
+      membershipId,
+      hasMembership,
+      purchaseString: cancellationString,
+      populateMembershipData,
+    });
+  };
+
+  function convertPriceToString(price) {
+    const interval = price.recurring.interval;
+
+    if (interval === 'year') {
+      return 'Yearly';
+    } else if (interval === 'month') {
+      return 'Monthly';
+    }
+  }
+
+  function capitalizeWord(string) {
+    return string.charAt(0).toUpperCase() + string.slice(1);
+  }
+
+  function buildCurrencyDisplay(priceObject) {
+    let currencySymbol;
+    if (priceObject.currency === 'eur') {
+      currencySymbol = '€';
+    } else if (priceObject.currency === 'usd') {
+      currencySymbol = '$';
+    }
+
+    const currency = priceObject.currency.toUpperCase();
+
+    return currency + ' ' + currencySymbol;
+  }
+
+  const urlSearchParams = new URLSearchParams(window.location.search);
+  const params = Object.fromEntries(urlSearchParams.entries());
+
+  const { interval, plan, pageLocation } = params;
+
+  const planValue = params.plan;
+
+  // add a bit of a delay otherwise it's a bit jarring
+  let timeoutValue = 300;
+  if (pageLocation === 'confirmPage') {
+    timeoutValue = 300;
+  }
+
+  // description to be shown under plan name
+  function getPlanDescription(plan) {
+    if (plan === 'Premium') {
+      return 'Badge on profile, Early Access to new features';
+
+      // if there's more plans added this needs to be expanded
+    } else {
+      return 'All Premium features, and No Ads';
+    }
+  }
+
+  if (!stillWaitingFromBackend && planValue && cardSaved) {
+    setTimeout(function () {
+      // clear query params
+      window.history.replaceState(null, null, window.location.pathname);
+
+      setHasShownModal(true);
+
+      // open confirm purchase
+      // $FlowFixMe
+      document.querySelector('[plan="' + plan + '"][interval="' + interval + '"]').click();
+    }, timeoutValue);
+  }
```

Comment:
⚠️ Render functions must be "pure". 
- `setTimeout` and other side-effects should not be called in a render function.

Reply:
Why is this marked resolved when it isn't ... did I miss something?

<img width="565" alt="image" src="https://user-images.githubusercontent.com/64950861/157377027-4ea32dc7-b5b8-4a78-bdb6-35ac56c85892.png">
https://reactjs.org/docs/hooks-reference.html#useeffect
https://gist.github.com/sebmarkbage/75f0838967cd003cd7f9ab938eb1958f

----

And is it not possible to force-click the element the React way?
https://stackoverflow.com/a/56903946/977819